flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1365,8 +1365,14 @@ IntrinsicLibrary::genIndex(mlir::Type resultType,
   // Call the descriptor-based Index implementation
   auto string = builder.createBox(loc, args[0]);
   auto substring = builder.createBox(loc, args[1]);
+  auto makeRefThenEmbox = [&](mlir::Value b) {
+    auto temp = builder.createTemporary(loc, builder.getI1Type());
+    auto castb = builder.createConvert(loc, builder.getI1Type(), b);
+    builder.create<fir::StoreOp>(loc, castb, temp);
+    return builder.createBox(loc, temp);
```

Comment:
Should `createBox` create the temp if needed when creating a box for an extended value that is not in memory instead of dealing with this here ? I am not sure this is useful in other places, but it looks like it might and I think it would be nice if `createBox` is always safe to call.

Reply:
I wondered the same thing. This was just the less invasive approach, but it might be better the other way.